src/exportToHubble.js

```diff
@@ -0,0 +1,83 @@
+const fs = require('uxp').storage.localFileSystem;
+const shell = require('uxp').shell;
+
+const CANCELLED = 'reasonCanceled';
+
+function showDialog(defaultInput = '') {
+    document.body.innerHTML = `
+      <style>
+        form {
+          width: 400px;
+        }
+        label {
+          display: flex;
+        }
+        .textfield {
+          flex: 2;
+        }
+        .h1 {
+          align-items: center;
+          justify-content: space-between;
+          display: flex;
+          flex-direction: row;
+        }
+        .icon {
+          border-radius: 4px;
+          width: 24px;
+          height: 24px;
+          overflow: hidden;
+        }
+      </style>
+      <dialog id="dialog">
+        <form method="dialog">
+          <h1>Trigger URL</h1>
+          <label>
+            <span>URL</span>
+            <input class="textfield" uxp-quiet="true" type="text" id="url" value=${defaultInput} placeholder="Your URL, e.g. https://cloudfunctions.net/export"/>
+          </label>
+          <footer>
+            <button id="cancel">Cancel</button>
+            <button type="submit" id="ok" uxp-variant="cta">OK</button>
+          </footer>
+        </form>
+      </dialog>
+    `;
+
+    const dialog = document.getElementById('dialog');
+
+    const cancelButton = document.getElementById('cancel');
+    cancelButton.addEventListener('click', () => dialog.close(CANCELLED));
+
+    const okButton = document.getElementById('ok');
+    okButton.addEventListener('click', e => {
+        dialog.close(document.getElementById('url').value);
+        e.preventDefault();
+    });
+
+    return dialog.showModal();
+}
+
+async function exportToHubble() { 
+    const file = await fs.getFileForOpening({types:['xd']});
+    if (!file) return;
+    const filePath = file.nativePath;
+
+    const folder = await fs.getDataFolder();
+    let urlFile = null;
+    try {
+        urlFile = await folder.getEntry('cloudurl.txt');
```

Comment:
Wrapping this in a try-catch doesn't make sense if you are just going to repeat it again in the catch.

If this throws inside the first attempt, why would you reexecute it in the catch block, just to have it throw again?

Reply:
Those are 2 different functions. The initial call tries to get the file from the data folder. If the call fails (i.e. it doesn't exist) we create the file in the catch block

Reply:
Haha wow wasn't paying attention